src/clj_puppetdb/core.clj

```diff
@@ -55,9 +53,9 @@
   contains additional metadata. Currently the only supported kind of metadata is:
   - :total (the total number of records available)"
   ([client path params]
-    (query-with-metadata client path nil params))
+   (query-with-metadata client path nil params))
   ([client path query-vec params]
-   (let [merged-params (merge params
+   (let [merged-params (merge {} params
```

Comment:
The precondition which fails is this one: https://github.com/mruzicka/clj-puppetdb/blob/c937dcc7144b12803bffe3cfd0226b6025350beb/src/clj_puppetdb/http.clj#L58